### ISSUE TITLE
[10] DenseHybrid: Hybrid Anomaly Detection for Dense Open Set Recognition

### ISSUE THREAD

**Issue by sy00n** [sy00n/DL_paper_review#11]
## 0. Abstract

- **기존 Anomaly detection에 존재하는 한계**
    
    Anomaly detection은 크게 _regular data로 generative modeling_하는 방법, _negative training data에 대해 discriminating 하는 방법_으로 구분지어 생각할 수 있다. 그런데 이 두가지 접근은 각각 다른 failure mode를 가진다. 따라서 이 둘을 합치는 하이브리드 알고리즘을 연구 목표로 한다. 
    
- **기존 hybrid approach가 부적절했던 이유와 개선법**
    
    Dense anomaly detection의 경우 1) translational equivariance와 2)very large input resolutions가 요구되기에 기존 하이브리드 접근 방식은 부적절하다. 따라서 우리는 logarithm of discriminative logits를 unnormalized joint distribution p^(x,y)으로 재해석해서 새로운 하이브리드 알고리즘을 설계한다. 
    

본 논문은 3개의 dense prediction을 recover하는 shared convolutional representation을 기반으로 한다. 

1.  **The closed-set class posterior P(y|x) (기존 방식)**
2.  **The dataset posterior P(d_in|x) (추가)**
3.  **Unnormalized data likelihood P^(x)  (추가)**

이중에서 2,3번은 standard training data, generic negative dataset 둘 다에서 학습된다. 

이렇게 해서 얻은 두 prediction에 의해 하이브리드 anomaly score를 산출해서 natural image에 대해 dense open set recognition을 수행할 수 있다. 

- **Loss function**
    
    custom loss 설계를 통해 backpropagation 시에 다루기 힘든 normalizing constant Z(θ)를 피함.
    
- **Evaluation Metrics**
    
    Open-mIoU(dense open-set performance를 위한 새로운 평가지표) 제시함.
    

## 1. Introduction

- 해결해야 할 문제
    
    현대 딥러닝의 standard training procedure은 closed-set이기 때문에 심각한 안전 문제를 일으킬 수 있다. 예를 들어 unknown object가 도로에 등장했을 때 misclassification하게 되면 자율주행 자동차는 심각한 사고로 이어질 것이다. 이런 위험은 dense anomaly detection으로 semantic segmentation을 보완하여 완화할 수 있다. 실제 세계에서는 dense open-set recognition은 anomalous pixel에 대해 결정을 거부할 수 있기 때문에 더욱 적합하다. 
    

![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/b61683c7-cd97-40dc-9689-6e77c94f65c2/Untitled.png)

- 기존 연구들과 그 한계점
    - Generative approaches (density estimation이나 image resynthesis를 기반으로 함)
        - inaccurately disperse the probability volume
        - rely on risky image resynthesis
    - Discriminative appraches (classification confidence, binary classifier, Bayesian inference를 기반으로 함)
        - 전체 input sapce에 대해 학습하는 것을 가정하기 때문에 unknown unknowns까지 포함될 수 있다.
        
- 본 논문에서 극복
    
    이 연구에서는 두 접근법을 hybrid anomaly detector로 합친다. 두 가지 추가적인 예측으로 standard semantic segmentation model를 보완한다. (closed set 하에서의 class posterior 뿐만 아니라 거기에 이 두개 추가)
    
    1. **Unnormalized dense data likelihood p^(x)**
    2. **Dense data posterior P(d_in|x)**
    
    1,2 번 두 예측은 모두 negative data를 써서 학습시킨다. 두 output을 join함으로써 정확하면서도 효과적인 dense anomaly detector DenseHybrid를 만든다. 
    

- Contribution
    1. Pixel-level에서 end-to-end learning을 가능하게 하는 최초의 하이브리드 anomaly detector를 제안한다. 
    2. likelihood evaluation, off-the-shelf negative dataset을 활용한 discrimination 이 둘을 합친다. 
    3. 우리의 실험은 minimal computational overhead임에도 불구하고 정확한 이상 탐지를 한다. 
    4. novel performance metric open-mIoU로 sota 달성한다.
    
    ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/1c2fb8c0-3e08-4e59-a5cb-0c4d76ebfda1/Untitled.png)
    

## 2. Related Work

 Training data의 생성 과정에서 벗어난 데이터를 탐지하는 것은 수십년 된 문제이다.  machine learning 커뮤니티에서는 anomaly detection 또는 Out-of-distribution detection으로도 알려져 있다. 

- Early Image-wise approach
    
    초기 image-wise approach에서는 max-softmax probability, input perturbations, ensembling, Bayesian uncertainty 등을 사용했다. 이후에는 broad negative dataset을 사용하거나 생성 기반 모델들로 학습시킨 데이터로부터 구분되게 학습하는 방식들 등이 좋은 성능을 보였다. 
    

- Generative approach
    
     또 하나의 접근 방식은 **Generative model에서 likelihood를 추정**하는 것이다. 이 방식은 anomalies가 inliers보다 더 높은 likelihood를 가질 수 있다는 문제가 있다. 추가 연구에 따르면 group-wise anomaly detection에서 더 나은 성능을 기대할 수 있지만 실질적으로 중요하지 않다. 생성 모델은 negative training data에서 낮은 likelihood를 할당하는 방향으로 학습할 수도 있는데, 이 방법이 probability volume의 sub-optimal dispersion 문제를 완화해준다. 
    

- Image-wise approach
    - 기존 생성 접근 방식은 dense likelihood estimate를 할 수 없었다. 반면 **max-softmax**나 **negative data로 discriminative training을 하는 방식**은 dense prediction에 쉽게 적용할 수 있다.
    - 많은 dense anomaly detector는 **일반 training data에 negative data를 붙여서 mixed-content images로 학습**된다.
    - standard classification head와 features를 공유하는 OOD head에 의해 discriminative anomaly detection이 가능해진다. Shared feature는 OOD performance를 개선하고 baseline semantic segmentation model과 관련해 neglectable compuational overhead를 발생시킨다는 장점이 있다.
    - 최근 연구에서는 negative pixel에 대해 large softmax entropy를 갖게 하는 연구도 있다. (Entropy maximization and meta classification for out-of-distribution detection in semantic segmentation. In: International Conference on Computer Vision, ICCV (2021)

- Other approach
    - Anomalies는 **feature space**에서도 recognize될 수 있지만 **subsampled된 feature representation** 및 **feature collapse**로 인해 **작은 물체 감지**를 잘 못한다.
    - 이전 접근 방식과 달리 **input과 resynthesised image 사이의 dissimilarity에 따라 anomaly detector를 구현**할 수 있다. 그러나 이 방법은 도로와 같은 균일한 배경에만 적잡하고 상당한 계산 오버헤드로 인해 실시간성이 떨어진다는 문제가 있음.
    
- Our approach
    - 본 논문의 접근은 Discriminative detection과 likelihood evaluation의 hybrid combination이다. Discriminative OOD detection에 대한 기존 연구들도 있지만 그 접근들과는 달리 **likelihood testing와의 시너지 효과를 통해 discriminative OOD detection을 개선**한다.
    - dense likelihood evaluation은 generative model을 discriminative feature에 fitting시키는 방식으로 수행되었다. 하지만 이 접근 방식은 two-phase training으로 인해 feature collapse에 취약하다. 또 작은 이상의 감지는 subsampling으로 인해 불안정해진다.
    - 따라서 기존 접근방식과는 달리 우리는 **standard dense prediction model과 full resolution에서의 anomaly detection의 joint training을 수행**한다.
    - 우리는 **logits을 unnormalized joint likelihood**로 재해석하여 dense likelihood evaluation을 수행한다.
    - 이전 연구에서도 (Your classifier is secretly an energy based model and you should treat it like one., ICRL 2020) unnorlized joint likelihood 방식을 적용했는데 이 방법은 large resolution에서 Langevin sampling의 intractability로 인해 dense prediction에 적합하지 않다. 따라서 우리는 mixed-content images에서 학습시키는 방식으로 재해석하고 intractable normalizing constant Z(θ)를 통해 역전파가 쉽도록 loss를 단순화했다.
    - 또 **hybrid anomaly score를 thresholding**하고 **standard semantic segmentation prediction과 결합**하여 open set recognition model을 설계했다.
    - 그리고 기존 dense recognition performance에 대한 standard metrics가 anomaly sample의 정확도는 고려하지 않는다는 점에 주목했고 또 기존 anomalous pixel의 presence를 discrimination하는 연구들은 recognition 성능보다 알고리즘의 robustness에 더 중점을 두었는데, 본 논문은 이전의 image-wise metrics와 유사하게 inlier와 outlier 모두에서 prediction quality를 계산하는 **novel anomaly-aware metric (open-mIoU)**를 제안한다.
